src/pages/pu-vendor-groups/index.js

```diff
@@ -0,0 +1,122 @@
+import Table from 'src/components/Shared/Table'
+import GridToolbar from 'src/components/Shared/GridToolbar'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useWindow } from 'src/windows'
+import { useResourceQuery } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import { PurchaseRepository } from 'src/repositories/PurchaseRepository'
+import VendorGroupsForm from './forms/VendorGroupsForm'
+
+const VendorGroups = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const { stack } = useWindow()
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50 } = options
+
+    try {
+      const response = await getRequest({
+        extension: PurchaseRepository.VendorGroups.qry,
+        parameters: `_startAt=${_startAt}&_pageSize=${_pageSize}&_filter=&_sortField=`
+      })
+
+      return { ...response, _startAt: _startAt }
+    } catch (error) {}
+  }
+
+  const {
+    query: { data },
+    labels: _labels,
+    invalidate,
+    paginationParameters,
+    refetch,
+    access
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: PurchaseRepository.VendorGroups.qry,
+    datasetId: ResourceIds.VendorGroups
+  })
+
+  const columns = [
+    {
+      field: 'reference',
+      headerName: _labels.reference,
+      flex: 1
+    },
+    {
+      field: 'name',
+      headerName: _labels.name,
+      flex: 1
+    },
+    {
+      field: 'nraName',
+      headerName: _labels.nra,
+      flex: 1
+    }
+  ]
+
+  const add = () => {
+    openForm()
+  }
+
+  const del = async obj => {
+    try {
+      await postRequest({
+        extension: PurchaseRepository.VendorGroups.del,
+        record: JSON.stringify(obj)
+      })
+      invalidate()
+      toast.success(platformLabels.Deleted)
+    } catch (error) {}
+  }
+
+  function openForm(recordId) {
+    stack({
+      Component: VendorGroupsForm,
+      props: {
+        labels: _labels,
+        recordId: recordId,
+        maxAccess: access
+      },
+      width: 600,
+      height: 500,
+      title: _labels.vendorGroups
+    })
+  }
+
+  const edit = obj => {
+    openForm(obj?.recordId)
+  }
+
+  return (
+    <VertLayout>
+      <Fixed>
+        <GridToolbar onAdd={add} maxAccess={access} />
+      </Fixed>
+      <Grow>
+        <Table
+          columns={columns}
+          gridData={data}
+          rowId={['recordId']}
+          onEdit={edit}
+          onDelete={del}
+          isLoading={false}
+          pageSize={50}
+          refetch={refetch}
+          paginationParameters={paginationParameters}
+          paginationType='api'
```

Comment:
here paginationtype should be client, and i guess we don't need paginationParameters anymore, pls check